internal/plugin/convert/function.go

```diff
@@ -0,0 +1,64 @@
+package convert
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/opentofu/opentofu/internal/providers"
+	"github.com/opentofu/opentofu/internal/tfplugin5"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func ProtoToCtyType(in []byte) cty.Type {
+	var out cty.Type
+	if err := json.Unmarshal(in, &out); err != nil {
+		panic(err)
+	}
+	return out
+}
+
+func ProtoToTextFormatting(proto tfplugin5.StringKind) providers.TextFormatting {
+	switch proto {
+	case tfplugin5.StringKind_PLAIN:
+		return providers.TextFormattingPlain
+	case tfplugin5.StringKind_MARKDOWN:
+		return providers.TextFormattingMarkdown
+	default:
+		panic(fmt.Sprintf("Invalid text tfplugin5.StringKind %v", proto))
+	}
+}
+
+func ProtoToFunctionParameterSpec(proto *tfplugin5.Function_Parameter) providers.FunctionParameterSpec {
+	return providers.FunctionParameterSpec{
+		Name:               proto.Name,
+		Type:               ProtoToCtyType(proto.Type),
+		AllowNullValue:     proto.AllowNullValue,
+		AllowUnknownValues: proto.AllowUnknownValues,
```

Comment:
I saw we have validations around `AllowNullValue` and `VariadicParameter`. But I didn't see any mention of `AllowUnknownValues`. We don't need any special logic for it?

Reply:
That's taken care of by cty/function.Function at the moment.  Once we have e2e testing, we will validate that assumption.